src/network/local_socket_reply.cpp

```diff
@@ -141,36 +139,113 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
     // Build the HTTP method part
     http_data += op;
     http_data += ' ';
-    http_data += request.url().toString();
+    http_data += request.url().path();
+
+    if (request.url().hasQuery())
+    {
+        http_data += "?";
+        http_data += request.url().query();
+    }
+
     http_data += " HTTP/1.1\r\n";
 
     // Build the HTTP Host header
-    // Host can be anything, so we'll use 'multipass'
-    http_data += "Host: multipass\r\n";
+    auto host = request.url().host();
+    if (!host.isEmpty())
```

Comment:
Is a request that doesn't have a host valid?

Reply:
No, it's not :wink:   Will either throw an exception or emit an error in `LocalSocketReply`.